webcompat/api/endpoints.py

```diff
@@ -72,10 +72,14 @@ def proxy_issues():
     params = request.args.copy()
 
     # If there's a q param, then we need to use the Search API
-    # and load those results. Unfortunately, we restrict search requests
-    # to logged in users.
+    # and load those results. For logged in users, we handle this at the
+    # server level.
     if g.user and params.get('q'):
         return get_search_results(params.get('q'), params)
+    # Non-authed users should never get here--the request is made to
+    # GitHub client-side)--but return out of paranoia anyways.
+    elif params.get('q'):
+        return
```

Comment:
heh but [501 Not implemented](http://tools.ietf.org/html/rfc7231#section-6.6.2) is more for method not implemented. :) It's more a [404 Not Found](http://tools.ietf.org/html/rfc7231#section-6.5.4) Subtleties of HTTP. Nothing prevent to send an appropriate message with that. The 404 is for the client, the body for the user. :)